python/GafferUI/CompoundEditor.py

```diff
@@ -121,6 +121,13 @@ def _createDetachedPanel( self, *args, **kwargs ) :
 
 		panel = _DetachedPanel( self,  *args, **kwargs )
 		panel.__removeOnCloseConnection = panel.closedSignal().connect( lambda w : w.parent()._removeDetachedPanel( w ) )
+
+		scriptWindow = self.ancestor( GafferUI.ScriptWindow )
+		if scriptWindow :
+			panel.setTitle( scriptWindow.getTitle() )
+			weakSetTitle = Gaffer.WeakMethod( panel.setTitle )
```

Comment:
I think @johnhaddon would generally declare the function inline on the next line rather than have this temporary variable.

Reply:
Yeah, we'd been chatting about this a wile back.  My understanding was that it was on the basis of python not having `const`, so it's harder to infer the use of the what is held in the temporary var. As this is an object rather than a value I wondered if it was more palatable. To me, calling `ancestor` three times in a row increases the line lengths at the cost of readability, `129` is bad enough imho.

Reply:
Oh come now, its no where near as bad as `485` which does an inline `Gaffer.WeakMethod` as well... Anyhow, I don't mind, I'll leave it to John.

Reply:
totally :) but didn't want to scope creep the commit 😂